peerconnection.go

```diff
@@ -55,6 +56,8 @@ type PeerConnection struct {
 	lastOffer  string
 	lastAnswer string
 
+	sdpOrigin *sdp.Origin
```

Comment:
Could we pull this from the `currentLocalDescription` or `pendingLocalDescription`? Would love to find a way to not add more state.

Reply:
using values from `currentLocalDescription` or `pendingLocalDescription` could somehow replace `PeerConnection.updateSDPOrigin()` method;

however, this way doesn't fully follow the spec for _session version_:
>The value of the generated \<session-version\> 
is _independent_ of the \<session-version\> of the
      _current local description_; in particular, in the case where the
      current version is N, an offer is created and applied with version
      N+1, and then that offer is rolled back so that the current
      version is again N, the next generated offer will still have
      version N+2.

before calling `SetLocalDescription()` (or `setDescription()` precisely) `currentLocalDescription` or `pendingLocalDescription` are still `nil`; and session version should be incremented by one regardless _current_ local description's version.

And currently these methods in `PeerConnection` are not well protected by lock:
- `CreateOffer`
- `CreateAnswer`
- `AddTransceiverFromKind`
- `AddTransceiverFromTrack`

But of course the lock issues could be solved later.

Reply:
:+1: ok yea I think one extra field is not a big deal at all considering what else we need to untangle.

I do feel strongly about the use of `unsafe` though.

Reply:
I removed the `unsafe` usage :)